src/models/user.js

```diff
@@ -4,8 +4,8 @@ module.exports = User;
 
 function User(attr) {
 	// TODO: Remove this
-	attr.name = attr.nick;
-	attr.mode = attr.modes[0] || "";
+	attr.name = attr.name || attr.nick;
+	attr.mode = attr.mode || (attr.modes && attr.modes[0]) || "";
```

Comment:
It's not an easy change because it will affect a lot of files. I'd rather tackle it in a separate PR because this one is getting pretty big already.